server/backend/api/app/routers/router_user_role_assignment.py

```diff
@@ -62,8 +71,30 @@ def delete_user_role_assignment(
 
     # Enforce application-level security
     application_id = crud_application.get_application_id_by_user_role_xref_id(
-        db, user_role_xref_id)
+        db, user_role_xref_id
+    )
     jwt_validation.authorize_by_app_id(application_id, db, token_claims)
 
+    # Enforce self-grant guard
+    target_user = crud_user.get_user_by_user_role_xref_id(db, user_role_xref_id)
+    enforce_self_grant_guard(db, requester, target_user.user_name, target_user.user_type_code)
+
     crud_user_role.delete_fam_user_role_assignment(db, user_role_xref_id)
     LOGGER.debug(f"User/Role assignment deleted successfully, id: {user_role_xref_id}")
+
+
+def enforce_self_grant_guard(db, requester, target_user_name, target_user_type_code):
+    requesting_user = crud_user.get_user_by_cognito_user_id(db, requester)
```

Comment:
Perhaps better to add some comment on why the value of "requester" is <= "username" is <= "cognito_user_id".